erpnext/public/js/controllers/transaction.js

```diff
@@ -310,6 +320,23 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 						if(!r.exc) {
 							me.frm.script_manager.trigger("price_list_rate", cdt, cdn);
 							me.toggle_conversion_factor(item);
+							if(show_batch_dialog) {
+								var d = locals[cdt][cdn];
+								$.each(r.message, function(k, v) {
+									if(!d[k]) {
+										d[k] = v;
+									}
+								});
+								let serial_no_batch_selector = new erpnext.SerialNoBatchSelector({
```

Comment:
create a method `erpnext.show_serial_batch_selector` and put `frappe.require` inside it.